c2corg_ui/static/js/gpxupload.js

```diff
@@ -66,6 +66,20 @@ app.GpxUploadController.prototype.importGpx_ = function(gpx) {
   var features = gpxFormat.readFeatures(gpx, {
     featureProjection: 'EPSG:3857'
   });
+  if (features.length > 1) {
+    var parser = new jsts.io.OL3Parser();
+    var gf = new jsts.geom.GeometryFactory();
+
+    var jstsFeatures = [];
+    for (var i = 0; i < features.length; i++) {
+      var geofeat = features[i].getGeometry();
+      var jstsfeat = parser.read(geofeat);
+      jstsFeatures.push(jstsfeat);
+    }
+    var jstsFeatureColl = gf.createGeometryCollection(jstsFeatures);
```

Comment:
For routes and outings, either a `LineString` or `MultiLineString` is expected. So, the API will complain about a geometry collection.

https://github.com/c2corg/v6_api/blob/c4152e4ffd3f2ff801cebc60dcec4eb310db6e35/c2corg_api/models/route.py#L328-L329

Reply:
The geometry collection goes through a union, so it should be good no?

Reply:
I am not sure from the docs: https://bjornharrtell.github.io/jsts/doc/api/jsts.geom.GeometryCollection.html#union